proposals/web-api-add-edit-with-media.md

```diff
@@ -0,0 +1,618 @@
+
+### Data Structure
+These are the required fields to represent the upload process for media.  The data dictionary change set can be found [here](https://docs.google.com/spreadsheets/d/1QGAe3hFvKHu9fMVzlGbI4gIuiSLjdTHO7leGK47bP88/edit?usp=sharing):
+
+```xml
+<EntityType Name="Media" hasStream="True">
+  ...
+  <Property Name="MediaStatus" Type="Edm.String" />
+  <Property Name="MediaStatusDescription" Type="Edm.String" />
+  ...
+</EntityType>
+```
+
+The `Media` resource/model will be annotated to have an OData stream attached.  
+
+### Status of the Media
+
+The status of the media will be tracked on the `Media` resource.  The `MediaStatus` field will show the status of the media.
+
+* `Incomplete` - the `Media` record that does not have a complete record 
+* `Processing` - the `Media` record is syntactically complete and is currently being processed by the backend
+* `Complete` - the `Media` record is validated and ready for consumption
+* `Deleted` - the `Media` record is no longer published 
+* `Rejected` - the `Media` has failed post processing for some reason.  Details of the problem will be presented in the `MediaStatusDescription` field.
+
+Some examples of `Rejected` status reason could be:
+* Media stream does not match `MediaType`  (Sent a video for a photo Media record)
+* The internal image checksum failed.
+* Image failed copyright ID check
+* Content of the media does not meet compliance rules
+
+## Media Add/Edit Process
+
+The creation of the media resource will be done using as a standard RESO Web API 2.1.0 Add/Edit process. The post SHOULD include all known resource record fields for the media with the exception of only the media byte stream.
+
+The response object MAY contain `mediaReadLink` and `mediaEditLink` annotations which the client MUST respect as per the OData specification when communicating the byte stream to the server. If fields are not provided the implicit URLs per the OData specification MUST then be used.  
+
+Byte stream endpoints MUST only expect previously established security credentials if they are in the same HTTP security domain as the original request.  Byte stream endpoints outside the original OData security domain cannot expect any security credentials to be provided outside of the contents of the respective annotation being used.
+
+The Byte stream endpoint will provide HTTP response code values related to the success of the byte stream POST.  All HTTP 2xx response codes are successful.  All HTTP 4xx error codes except for HTTP 409 (Conflict) SHOULD have the client query the OData resource to validate/refresh the annotation URL and retry the POST using the URL.
+
+The HTTP 409 (Confict) response designates that the byte stream cannot be replaced.  This can happen if the implementation is using a write-once model for media to retain history.  When a 409 is received, the client SHOULD add a new media record with the updated byte stream and then delete the original Media record.
```

Comment:
Might want to change the language a bit here to say that "Providers MUST return an HTTP 409 (Conflict) status they are using a write-once model and the client tries to update a record..."